### PR TITLE
Store complete meta on worker and update supervisor meta via fetching from workers

### LINKED ISSUE #2907: [BUG] Mars serialization took too much time
**Describe the bug**
When executing following task with 3000 parallelism, mars took 2.5h to finished using 55 workers each has 9 subpools. But the serialization of supervisor took 2.2h in it:
```python
df = md.DataFrame(
    mt.random.RandomState(0).rand(3000_000, 5, chunk_size=1000),
    columns=list('abcde'))
df.groupby(['a', 'b']).apply(lambda df: [1,2]).reset_index().execute()
```
image
mars_tests_mubai_o1

**To Reproduce**
To help us reproducing this bug, please provide information below:
1. Your Python version: python 3.7.9
2. The version of Mars you use: master
3. Versions of crucial packages, such as numpy, scipy and pandas
4. Full stack of the error.
5. Minimized code to reproduce the error.
```python
df = md.DataFrame(
    mt.random.RandomState(0).rand(3000_000, 5, chunk_size=1000),
    columns=list('abcde'))
df.groupby(['a', 'b']).apply(lambda df: [1, 2]).reset_index().execute()
```

**Expected behavior**
We need to:
* Find out what is being sent to the supervisor
* reduce the data sent to mars supervisor
* Speed up mars serialization performance.

**Additional context**

### LINKED ISSUE #2908: Enable worker meta to store large chunk meta like dtypes
## Background

Now, all chunk meta including dtypes, index_value, columns_value etc will be stored into supervisor meta, this overhead may be significant when number of chunks is large.

## Solutions

Worker meta could be enabled to store large chunk meta like dtypes, index_values etc, and supervisor meta would only record bands, memory_size etc.

Besides, not every chunk meta should be recorded, we need to only record those chunks whose index contains 0, like (0, 1), (1, 0), this will reduce serialization significantly.

## Tasks

* Support worker meta service (#2909)
* Store chunk meta into worker meta, for supervisor, update meta that fetched from workers.

### PR DESCRIPTION
**Pull request by qinxuye**
## What do these changes do?

This PR support storing complete meta on worker and update supervisor meta via fetching from workers.

## Related issue number

Fixes #2908 
#2791 

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

File: mars/tests/core.py [hunk 1]
```diff
@@ -355,9 +355,12 @@ def assert_dataframe_consistent(self, expected, real):
...
                 # ignore check when columns length is nan or dtypes undefined
-                pd.testing.assert_index_equal(
-                    expected.dtypes.index, self.adapt_index_value(real.dtypes.index)
-                )
+                try:
+                    pd.testing.assert_index_equal(
+                        expected.dtypes.index, self.adapt_index_value(real.dtypes.index)
+                    )
+                except AssertionError:
+                    raise
```
Comment (hekaisheng) [inline mars/tests/core.py:363]: What does this change do?

Comment (qinxuye) [reply mars/tests/core.py:363]: Oh, should be reverted.

**Review by hekaisheng — APPROVED** [review]
LGTM

**Review by wjsi — APPROVED** [review]
LGTM

**Pull request merged**